[docs] Add GraphQL-spec error codes (#8150)
apollographql/router: docs/source/routing/errors.mdx

</Note>

## GraphQL spec error codes

These error codes are defined in the [GraphQL spec](https://spec.graphql.org/October2021/) and can appear in client responses under `errors[].extensions.code`, which is an established convention found in [GraphQL error extensions](https://spec.graphql.org/October2021/#sec-Errors.Error-result-format). Because these error codes are part of the core spec, they cannot be disabled directly as they are part of the core Router implementation. Only Apollo Router-specific features or subgraph errors can be disabled.

> Changing these errors makes your router non-compliant with the GraphQL spec. To edit the response, you must use an external proxy.

Reviewer: @smyrick the last sentence is unclear to me. What do we mean by "edit the response"? Can we link out to anything there for more information?

> Changing these errors makes your router non-compliant with the GraphQL spec, and there currently is not a way to mutate the HTTP payload to remove these errors with the router. If you need that behaviour it will need to be done via external system like an HTTP-proxy layer.

Reviewer: ```suggestion
Changing these errors makes your router non-compliant with the GraphQL spec. If you need to change these errors, you need to use an external system, such as an HTTP proxy layer.
```
Simplifying + reducing redundant information from above paragraph.

Learn how to see these error codes in Studio with [extended error reporting](/graphos/routing/graphos-reporting#enabling-extended-error-reporting).

<PropertyList kind="errCodes">

<Property name="GRAPHQL_VALIDATION_FAILED">

The operation failed during GraphQL validation.

</Property>
<Property name="GRAPHQL_UNKNOWN_OPERATION_NAME">

The operation could not be executed because the operation name was invalid or
did not match an operation in the query document.

</Property>
<Property name="PARSING_ERROR">

The operation could not be parsed as GraphQL.

</Property>

</PropertyList>

## Apollo Router error codes

Additional error codes can appear in client responses under `errors[].extensions.code`, which is an established convention found in [GraphQL error extensions](https://spec.graphql.org/October2021/#sec-Errors.Error-result-format). Learn how to see these error codes in Studio with [extended error reporting](/graphos/routing/graphos-reporting#enabling-extended-error-reporting).

> These errors are part of Apollo Router and not the core GraphQL spec, so they can be suppressed or disabled with [router customizations](/graphos/routing/customization/overview).

Reviewer: ```suggestion
You can suppress or disable these error codes using [router customizations](/graphos/routing/customization/overview).
```
Simplifying. Since this is already under the "Apollo Router error codes" heading, I think it's safe to omit.
